src/model/book-search/book-search.saga.js

```diff
@@ -0,0 +1,20 @@
+import BOOKSEARCH from './book-search.types';
+import { getBookSearhTerm } from './../raw-selectors';
+import { call, take, select, put } from 'redux-saga/effects';
+import data from './../../api/data';
+import { onFetchBooks } from './../books/books.actions';
+
+function *bookSearchAttempt() {
+  while (true) {
+    yield take(BOOKSEARCH.SEARCH_ATTEMPTED);
+    const searchTerm = yield select(getBookSearhTerm);
+    const searchedBooksResult = yield data.getSearchRemoteBooks(searchTerm)
+    yield put(onFetchBooks(searchedBooksResult));
+  }
+}
+
```

Comment:
Saga middleware

- line 8: since sagas use js generators you can run a never ending loop that "waits"
- line 9: "listens" for the `BOOKSEARCH.SEARCH_ATTEMPTED` action to get fired then moves to the next line
- line 10 asynchronously selects a property from the Redux store using saga `select` and stores that in `searchTerm`
- line 11 preforms an asynchronous api request to get searched books and stores that in `searchedBooksResult`  
- line 12: `put` is another way of firing a Redux action. This line first the `onFetchBooks` actions and passes in the `searchedBookResults` that redux can use.